pkg/cloud/services/ssm/secret_fetch_script.go

```diff
@@ -196,6 +196,7 @@ if [ ${GUNZIP_RETURN} -ne 0 ]; then
 fi
 
 log::info "restarting cloud-init"
-systemctl restart cloud-init
+rm -rf /var/lib/cloud/instances
+cloud-init clean --reboot
```

Comment:
```suggestion
reboot
```

Reply:
im going to try again with some updates i made to failing tests https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5118

Reply:
@SriRamanujam the tests are all passing - mind building and trying again locally?

Reply:
We also hit this issue and we found out that reboot is not needed after `cloud-init clean`. We just needed to restart all cloud-init services in the order.

```suggestion
rm -rf /var/lib/cloud/instances
cloud-init clean
systemctl restart cloud-init-local
systemctl restart cloud-init
systemctl restart cloud-config
systemctl restart cloud-final
```